src/include/threadpool/mono_queue_pool.h

```diff
@@ -24,7 +24,9 @@ namespace threadpool {
  */
 class MonoQueuePool {
  public:
-  MonoQueuePool(int task_queue_size = 32, int worker_pool_size = 4)
+  MonoQueuePool() = delete;
```

Comment:
I don't think you need this because you've defined a constructor with arguments. This is just noise.